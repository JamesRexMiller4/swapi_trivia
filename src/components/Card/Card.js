import React from 'react';
import './Card.scss';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

const Card = (props) => {
// For character cards the props being spread out and passed down will need to be grabbed
// as {id, name, homeworld, species}

  // homeworld, population of homeworld is a url for another fetch request
  // species is another fetch request
  // films is another fetch request
    let card;
    props.type === 'movie' ? card =
      <div className="wrapper-card-div">
        <ul className="card-details-ul">
          <li><strong className="episode-title">Episode {props.id}:</strong> {props.title}</li>
          <li><strong>Release Year:</strong> {props.date}</li>
        </ul>
        <Link to={`/movies/${props.id}`} key={props.id}>
          <button className="character-btn">View Characters</button>
        </Link>
      </div> :
    card =
      <div className="wrapper-card-div">
        <ul className="card-details-ul">
          <li><strong>Name:</strong> Luke SkyWalker</li>
          <li><strong>Homeworld:</strong> Tattoiine</li>
          <li><strong>Population of Tatootine:</strong> 400 million</li>
          <li><strong>Species:</strong> Human</li>
          <li><strong>Other Films:</strong> all seven</li>
        </ul>
      </div>

    return (
      <section className="section-card">
      {card}
      </section>
    )
  }
// }

// For character cards
// return (
//   <section className="section-card">
//   <div className="wrapper-card-div">
//     <ul className="card-details-ul">
//       <li><strong>Name:</strong> Luke SkyWalker</li>
//       <li><strong>Homeworld:</strong> Tattoiine</li>
//       <li><strong>Population of Tatootine:</strong> 400 million</li>
//       <li><strong>Species:</strong> Human</li>
//       <li><strong>Other Films:</strong> all seven</li>
//     </ul>
//   </div>
// </section>
// )

Card.propTypes = {
  id: PropTypes.number,
  title: PropTypes.string,
  date: PropTypes.string,
  characters: PropTypes.array
}



export default Card
