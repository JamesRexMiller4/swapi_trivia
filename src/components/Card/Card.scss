.section-card {
  width: 500px;
  height: 200px;
  background-color: white;
  margin: 20px;
  border: 10px solid;
  color: purple;
}

// @mixin imagegrid($qty, $margin) {
//   width: ((100% - (($qty - 1) * $margin))/ $qty);
//   &:nth-child(n) {
//     margin-right: $margin;
//     margin-bottom: $margin;
//   }
//   &:nth-child(#{qty}n) {
//     margin-right: 0;
//     margin-bottom: 0;
//   }
// }