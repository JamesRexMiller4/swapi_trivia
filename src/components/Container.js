import React from 'react';
import './Container.scss';

const Container = () => {
  return (
    <section className="container-section">
    </section>
  )
}


export default Container;