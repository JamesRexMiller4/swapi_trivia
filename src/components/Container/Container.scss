@import '../variables.scss';

.container-section {
  background: {
    attachment: fixed;
    image: url('../../images/star-wars-background.jpg');
    repeat: no-repeat;
    size: cover;
  }
  display: block;
  height: 100vh;
  overflow-x: scroll;
  position: relative;
  width: 100%;
}

.grid-container {
  display: grid;
  grid-column-gap: 70px;
  grid-row-gap: 20px;
  grid-template-columns: 1fr 1fr 1fr;
  margin: {
    left: auto;
    right: auto;
  }
  width: 90%;
}
