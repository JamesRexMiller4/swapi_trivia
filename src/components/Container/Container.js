import React from 'react';
import PropTypes from 'prop-types';
import Card from '../Card/Card.js'
import './Container.scss';

const Container = (props) => {
  let cardData;

  if (props.data.type !== 'movie') {
  const tenToDisplay = props.data.slice(0, 10)
    cardData = tenToDisplay.map((data, index) => {
  return (<Card {...data} setPath={props.setPath} key={index} favorite={props.favorite}/>)

  })

  } else {
    cardData = props.data.map((data, index) => {
      return (<Card {...data} setPath={props.setPath} key={index}/>)
      })
    }

    return (
      <section className="container-section">
        <div className="grid-container">
          {cardData}
        </div>
      </section>
    )
}

Container.propTypes = {
  data: PropTypes.array,
  setPath: PropTypes.func, 
  favorite: PropTypes.func
}


export default Container;
