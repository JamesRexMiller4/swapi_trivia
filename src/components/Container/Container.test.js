import React from 'react';
import Container from './Container';
import Card from '../Card/Card'
import {shallow} from 'enzyme';

describe('Container', () => {
  let wrapper, data;

  it('should match the snapshot with all the movie data passed through', () => {
    data = [
    {key: 4, id: 4, title: 'A NEW HOPE', date: '1977', characters: ['Luke', 'RD-D2', 'Chewie'],
    openingCrawl: 'In the beginning, there was Jack', type: 'movie'}, 
    {key: 5, id: 5, title: 'THE EMPIRE STRIKES BACK', date: '1980', characters: ['Han', 'Vadar', 'C3PO'],
    openingCrawl: 'And Jack had a groove', type: 'movie'}, 
    {key: 5, id: 6, title: 'RETURN OF THE JEDI', date: '1983', characters: ['Yoda', 'RD-D2', 'Luke'],
    openingCrawl: 'And from this groove, came the groove of all grooves', type: 'movie'}, 
    {key: 5, id: 7, title: 'THE FORCE AWAKENS', date: '2015', characters: ['B-B8', 'Chewi', 'Snope'],
    openingCrawl: 'And while one day viciously throwing down on his box, Jack boldly declared LET THERE BE HOUSE', type: 'movie'}
  ]
    wrapper = shallow(<Container data= {data} />)

    expect(wrapper).toMatchSnapshot()
  })

});