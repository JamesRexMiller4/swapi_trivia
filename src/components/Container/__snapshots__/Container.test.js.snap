// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container should match the snapshot with all the movie data passed through 1`] = `undefined`;
